scripts/get_orphaned_functions.py

```diff
@@ -0,0 +1,243 @@
+import os
+import re
+import csv
+
+# Define Working Directory, i.e, '/pecan
+main_dir = os.getcwd()
+
+# Define a list of color codes for printing. THis proves to be useful for distinction during debugging
+color_codes = [
+    "\033[30m",  # BLACK-0
+    "\033[31m",  # RED-1
+    "\033[32m",  # GREEN-2
+    "\033[33m",  # YELLOW-3
+    "\033[34m",  # BLUE-4
+    "\033[35m",  # MAGENTA-5
+    "\033[36m",  # CYAN-6
+    "\033[37m",  # WHITE-7
+    "\033[90m",  # BRIGHT_BLACK-8
+    "\033[91m",  # BRIGHT_RED-9
+    "\033[92m",  # BRIGHT_GREEN-10
+    "\033[93m",  # BRIGHT_YELLOW-11
+    "\033[94m",  # BRIGHT_BLUE-12
+    "\033[95m",  # BRIGHT_MAGENTA-13
+    "\033[96m",  # BRIGHT_CYAN-14
+    "\033[97m",  # BRIGHT_WHITE-15
+]
+
+# Reset
+ENDC = "\033[0m"
+
+# Modules to ignore while checking for orphaned functions
+ignored_modules = ["data.hydrology", "DART"]
+
+# Function to get the list of modules
+def get_module_list():
+    module_dir = os.path.join(main_dir, "modules")
+    modules = os.listdir(module_dir)
+    return [module for module in modules if module not in ignored_modules and not module.startswith('.')]
+
+# List of modules to check for orphaned functions
+module_list = get_module_list()
+
+# Initialize two lists to store the orphaned functions
+# List_1: Purely orphaned functions
+# List_2: Functions that are utilized externally
+list_1 = []
+list_2 = []
+
+
+# Define a function to list all functions in a module
+def list_functions_in_module(module_path):
+    """
+    List all functions defined in the R files of a module.
+
+    Args:
+    module_path (str): The path to the module's R directory.
+
+    Returns:
+    list: A list of function names defined in the module.
+    """
+    function_pattern = re.compile(r"\b([a-zA-Z_]\w*(?:\.\w+)*)\s*<-\s*function")
```

Comment:
This won't catch `funcname = function(...`, which is legal R and my quick grep says we've used 258 times in this repo.

Reply:
> Note that this won't catch `funcname = function(...`

I guess this expression is only being utilized for our `lambda` functions. However, there still exists a possibility of these lambda's being exported to outer packages for utilization. But, as far as I seem to know, I haven't encountered any such lambda function being utilised externally (still does not rules out any such possibility).

Maybe changing our expression as such may help us in identification of such `lambda` functions: 
```r
function_pattern = re.compile(r"\b([a-zA-Z_]\w*(?:\.\w+)*)\s*(?:<-\s*function|\s*=\s*function)")
```

Do you have any other opinions?

Reply:
Not sure what you mean by "this expression is only being utilized for our lambda functions". I agree with @infotroph that `funcname = function(` is valid and needs to be searched for

Reply:
My bad. I got confused. I've added searching functionality for functional declarations of type `funcname = function(` as well. Can you review it as well?

Reply:
Also, I've filtered out all such occurances of `error = function(e) ....`